components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/util/TemplateManagerHelper.java

```diff
@@ -322,26 +321,61 @@ public static void validateTemplatesProperties(RuleTemplate ruleTemplate) throws
     /**
      * Checks whether a given Template file has valid content.
      * Validation criteria : //todo: confirm validation criteria for templates
-     * - name
-     * - maximumInstances
-     * - maxNumberOfNodes
-     * - javascript
-     * - siddhiApps
-     * - properties //todo: validate whether all templated elements are referred as properties?
+     * - type
+     * - content
+     * - ExposedStreamDefinition
      *
-     * @param template Given Template object
+     * @param ruleTemplate Given Template object
      * @throws TemplateManagerException
      */
-    public static void validateTemplate(Template template) throws TemplateManagerException {
-        //todo: no need mostly.
-    }
+    public static void validateTemplate(RuleTemplate ruleTemplate) throws TemplateManagerException {
+        Collection<Template> templates = ruleTemplate.getTemplates();
+        /**
+         * Validation for the
+         *
+         * **/
```

Comment:
Please fix the comment

Reply:
Fixed by b14dd18f396f799bd32670842869feb2475e778e